pkg/ottl/parser_collection_test.go

```diff
@@ -491,18 +476,17 @@ func Test_ParseConditions_MultipleContexts_Success(t *testing.T) {
 }
 
 func Test_ParseConditions_NoContextInferredError(t *testing.T) {
-	pc, err := NewParserCollection[any](component.TelemetrySettings{})
+	pc, err := NewParserCollection[any](componenttest.NewNopTelemetrySettings())
 	require.NoError(t, err)
-	pc.contextInferrer = &mockStaticContextInferrer{""}
 
 	conditions := mockGetter{values: []string{`bar.attributes["bar"] == "foo"`}}
 	_, err = pc.ParseConditions(conditions)
 
-	assert.ErrorContains(t, err, "unable to infer context from conditions")
+	assert.ErrorContains(t, err, "is not a supported context")
```

Comment:
Same as the above comment to keep the mock, they're different errors coming from different places.